decidim-core/spec/lib/nicknamizable_spec.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+module Decidim
+  describe Nicknamizable do
+    subject do
+      class DummyTestUser < ApplicationRecord
+        self.table_name = :decidim_users
+
+        extend Nicknamizable
+      end
+    end
+
+    describe "#nicknamize" do
+      it "copies non-duplicated usernames following slugization rules" do
+        expect(subject.nicknamize("peter")).to eq("peter")
+      end
+
+      it "slugizes non-duplicated usernames" do
+        expect(subject.nicknamize("Rodríguez San Pedro")).to eq("rodriguez_san_pedro")
+      end
+
+      it "trims very long usernames" do
+        expect(subject.nicknamize("Felipe Juan Froilan de todos los Santos"))
+          .to eq("felipe_juan_froilan_")
+      end
+
+      it "resolves conflicts with current nicknames" do
+        create(:user, nickname: "ana_pastor")
+
+        expect(subject.nicknamize("ana_pastor")).to eq("ana_pastor_2")
```

Comment:
RE: the comment in the class, imagine we already have `ana_pastor_2` up to `ana_pastor_4`. Will a new `ana_pastor` be moved to `ana_pastor_5`?

Reply:
No, in that case `ana_pastor` would be used since that name is available.

Reply:
Oh, yes, sorry, I meant with `ana_pastor` being taken too :D

Reply:
In that case, yes, that's the idea. Do you think it warrants another test case maybe?

Reply:
My doubt was with the `2.step` part of the class code, I'd have written this with a `while` 😅

Reply:
I used `2.step` because I wanted to start with `ana_pastor_2` instead of `ana_pastor_1`, since it looks kind of more correct... meh, I don't think the renamed users will be very happy anyways. :smile:

Reply:
Well, and because using `step` gives me the counter for free... :)

Reply:
Ah, I see! Nice!

Reply:
OK, so the problem was just me not understanding how the `step` method works... :sweat_smile:

Reply:
Aaaaah, I see. `step` on an array works differently, easily confused!